docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,299 @@
+- The content of mpu_regs[] is initialized inside exec_module function, using mpu_get_register_value function.
+- Same MPU regions are shared by all the apps.
+- MPU regions are set, using saved configurations, at the time of context switch to the app.
+- Below 2 cases arise for MPU region configuration for an app:
+	1. **CONFIG_OPTIMIZE_APP_RELOAD_TIME=y**
+		- In this case, 3 MPU regions are used for an app.
+		- Below is MPU region info for this case:
+		```
+		----------------------------------------------------------------------------
+		MPU Region		Memory Segments		Permissions
+		----------------------------------------------------------------------------
+		MPU_REG_NUM_APP_TXT	TEXT			ROX
+		MPU_REG_NUM_APP_RO	RO			RONX
+		MPU_REG_NUM_APP_DATA	Complete RAM partition	RWNX
+		----------------------------------------------------------------------------
+		```
+	2. **CONFIG_OPTIMIZE_APP_RELOAD_TIME=n**
+		- In this case, only 1 MPU region is used for entire RAM address range for an app.
+		- Below is MPU region info in this case:
+		```
+		----------------------------------------------------------------------------
+		MPU Region		Memory Segments		Permissions
+		----------------------------------------------------------------------------
+		MPU_REG_NUM_APP		Complete RAM partition	RWX
+		----------------------------------------------------------------------------
+		```
+
+#### Configuring MPU Regions for Common Binary
+- A Common Binary contains shared code b/w all apps. This feature is enabled with CONFIG_SUPPORT_COMMON_BINARY.
+- MPU Region settings are configured at time of loading common binary (inside load_binary function).
+- Below 2 cases arise for MPU region configuration for common binary:
+	1. **CONFIG_OPTIMIZE_APP_RELOAD_TIME=y**
+		- In this case, 3 MPU regions are used for common binary.
+		- Below is MPU region info for this case:
+		```
+		----------------------------------------------------------------------------
+		MPU Region			Memory Segments		Permissions
+		----------------------------------------------------------------------------
+		MPU_REG_NUM_COM_LIB_TXT		TEXT			ROX
+		MPU_REG_NUM_COM_LIB_RO		RO			RONX
+		MPU_REG_NUM_COM_LIB_DATA	Complete RAM partition	RWNX
+		----------------------------------------------------------------------------
+		```
+	2. **CONFIG_OPTIMIZE_APP_RELOAD_TIME=n**
+		- In this case, only 1 MPU region is used for entire RAM address range for common binary.
+		- Below is MPU region info in this case:
+		```
+		----------------------------------------------------------------------------
+		MPU Region		Memory Segments		Permissions
+		----------------------------------------------------------------------------
+		MPU_REG_NUM_COM_LIB	Complete RAM partition	RWNX
+		----------------------------------------------------------------------------
+		```
+
+#### Configuring MPU region for Stack Overflow Protection
```

Comment:
Because of this, we should re-organize the tabs.
I sent a mail to you with suggestion. Could you check that?

Reply:
Organised this section as per suggestion in mail. Please check.

Reply:
As it is not possible to split/merge table cells in markdown, I have added an image of powerpoint table in the doc.
Image is kept at : docs/media/MPU_Usages_of_platform_table.png